doc/rds-day-2.md

```diff
@@ -0,0 +1,153 @@
+
+In Parameter Groups, there are two types of parameters: dynamic and static.
+Dynamic parameters do not require a restart for their values to be applied to the running instance / cluster.  Static parameters require a restart for their values to be applied. Additionally, dynamic parameters support specifying how changes to them are applied. When `immediate` is specified the changes to dynamic parameters are applied immediately. When `pending-reboot` is specified, the changes to dynamic parameters are applied during next restart or during the next maintenance window, whichever is earlier. 
+
+Since static parameters do not support `immediate` apply option, specifying this in your composition could lead to some unexpected error. Therefore, extra care should be taken when exposing this resource to your end users. End users may not be aware of underlying engine specifications.
+
+Summarizing everything above effectively means there are a few general approaches to managing RDS configuration changes. 
+1. You want to ensure that parameter group values in running cluster / instance match what is defined in your Git repository with no delay. The only sure way to ensure that is restarting the cluster/ instance during the reconciliation process. 
+3. You can wait for parameter group changes to be applied during the next maintenance window. This means you may need to wait maximum 7 days for the changes to be applied. 
+4. The change does not have to be applied immediately but it needs to happen sooner than 7 days. This requires a separate workflow to restart cluster / instance. 
+
+
+
+For reference, problems encountered during parameter group update in ACK and Terraform are discussed in [this issue](https://github.com/aws-controllers-k8s/community/issues/869) and [this blog post](https://tech.instacart.com/terraforming-rds-part-3-9d81a7e2047f). 
+
+## Solutions
+
+### Check during PR
+Use Pull Request as a checkpoint and ensure developers are aware of potential consequences of the changes. An example process may look something like the following. 
+
+
+```mermaid
+flowchart LR
+    FieldDefinitions(Fields that need restarting)
+    StepFiles(Get changed files)
+    StepCheck(Check if fields that require restart changed)
+    Comment(Comment on PR)
+    FieldDefinitions <--reference--> StepCheck
+
+    PR(PR Created) --> CR(Does this need DB Restart) --need restart--> StepFiles --> StepCheck --> Comment --> Approval --> Merge
+    CR --no restart--> Approval
+```
+
+In this example, whenever a pull request is created, a workflow is executed and a comment is created on the PR warning the developers of potential impacts. When developers approve the PR, it implies that they are aware of consequences.
+To check if a PR is impacted, you can use of the following options:
+- Parse git diff and search for changes to "dangerous" fields
```

Comment:
can we provide a list of "dangerous" fields? or is it going to be subject to change and hard to maintain? it would be good for people to not have to navigate out to see the list, if possible.

Reply:
also we should define earlier that we would refer to such and such fields as "dangerous" fields. also maybe the "outlaw" or "unruly" fields? lolz

Reply:
LOL yeah I couldn't think of good term for it. 
"Dangerous" fields are documented within RDS documentation https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html
As far as I can tell, there's no built-in way of getting this information programmatically. 

I counted 9 fields for non-multi-AZ and 3 for multi-AZ. Perhaps it's okay to document it in this doc since numbers are low. 

As for parameter groups, ALL static parameters require a restart. As far as I can tell, you can get which values are static by calling `describe-engine-default-parameters`. This is engine dependent and possibly version dependent too. So you may end up maintaining quite a large number of fields. Good news is that you can easily automate it.

Reply:
Maybe we can add an Admission Webhook for when a static value is changed to warn that the change requires db restart and will be applied in the next maintenance window.

Reply:
Yeah something like that is fine as long as users understand the underlying RDS concepts and there is a mechanisms to restart instances on-demand. It's just that there is a disconnect between the actual state of DB and the intentions of the users. It's not in-sync when GitOps controllers say it is in-sync.